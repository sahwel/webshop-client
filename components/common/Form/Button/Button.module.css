@tailwind components;


@layer components {
    .button {
        @apply w-full p-3 bg-button-light rounded text-white drop-shadow-button-ligth
    }

    :global(.dark) .button{
        @apply bg-button-dark text-text-dark drop-shadow-button-dark
    }
    
}
    
