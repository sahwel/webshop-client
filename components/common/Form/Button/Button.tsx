import { ButtonHTMLAttributes, FunctionComponent } from 'react';
import styles from './Button.module.css';
import cl from 'classnames';
interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

const Button: FunctionComponent<ButtonProps> = ({
  className,
  children,
  ...props
}) => {
  return (
    <button className={cl(className, styles.button)} {...props}>
      {children}
    </button>
  );
};

export default Button;
