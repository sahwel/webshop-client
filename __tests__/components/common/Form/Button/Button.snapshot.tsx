import { render } from '@testing-library/react';
import Button from '@/components/common/Form/Button/Button';

describe('Button Snapshot', () => {
  it('renders SecondButtonSection unchanged', () => {
    const { container } = render(<Button>Save</Button>);
    expect(container).toMatchSnapshot();
  });
});
