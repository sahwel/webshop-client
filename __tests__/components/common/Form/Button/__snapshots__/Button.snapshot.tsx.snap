// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot renders SecondButtonSection unchanged 1`] = `
<div>
  <button
    class="button"
  >
    Save
  </button>
</div>
`;
